File: main/scala/mod/iceandshadow3/compat/CServerCmdSender.java
package mod.iceandshadow3.compat;

import net.minecraft.command.ICommandSender;
import net.minecraft.server.MinecraftServer;

/**
 * Wrapper for a ICommandSender+MinecraftServer pair.
 * Fairly incomplete; will be extended when need be.
 */
public class CServerCmdSender {
	final MinecraftServer server;
	final ICommandSender sender;
	
	public CServerCmdSender(MinecraftServer server, ICommandSender sender) {
		this.server = server;
		this.sender = sender;
	}
	public CServerCmdSender(ICommandSender sender) {
		this(sender.getServer(), sender);
	}
	public String getSenderName() {
		return sender.getName();
	}
	public boolean isAdmin() {
		return sender.canUseCommand(3, "");
	}
}

File: main/scala/mod/iceandshadow3/compat/CServerCmdSender.scala
package mod.iceandshadow3.compat;

import net.minecraft.command.ICommandSender;
import net.minecraft.server.MinecraftServer;

/**
 * Wrapper for a ICommandSender+MinecraftServer pair.
 * Fairly incomplete; will be extended when need be.
 */
class CServerCmdSender(
	private[compat] val server: MinecraftServer,
	private[compat] val sender: ICommandSender
) {
	def this(sender: ICommandSender) = this(sender.getServer(), sender)
	def getSenderName(): java.lang.String = sender.getName();
	def isAdmin(): Boolean = return sender.canUseCommand(3, "");
}
